src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -245,6 +254,14 @@ def __init__(self, algorithm):
     algorithm = utils.read_only_property("_algorithm")
 
 
+@utils.register_interface(EllipticCurveKeyExchangeAlgorithm)
+class ECDH(object):
```

Comment:
On Sun, 2015-05-10 at 02:02 -0700, Alex Stapleton wrote:
> > @@ -245,6 +254,14 @@ def __init__(self, algorithm):
> >      algorithm = utils.read_only_property("_algorithm")
> >  
> >  
> > +@utils.register_interface(EllipticCurveKeyExchangeAlgorithm)
> > +class ECDH(object):
> 
> Should this also take the curve?

The curve is implicit in the key used ?
Or should we explicitly provide a curve and then fail if the key has a
different curve ?

Simo.

-- 
Simo Sorce * Red Hat, Inc * New York